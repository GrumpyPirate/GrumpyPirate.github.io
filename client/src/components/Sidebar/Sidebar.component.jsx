import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import Sitenav from 'components/Sitenav/Sitenav.container';
import Container from 'components/Layout/Container/Container.component';
import ScreenreaderOnlyText from 'components/Accessibility/ScreenreaderOnlyText/ScreenreaderOnlyText.component';

import classes from './Sidebar.component.scss';

const Sidebar = ({ isMobileNavigationOpen, toggleMobileNavigation }) => (
  <aside className={classes['sidebar']}>
    <Container>
      <div className={classes['sidebar__content']}>
        <Link to="/" className={classes['sidebar__brand']}>
          <figure className={classes['sidebar__brand__image']} alt="" width="52" height="52" />
          <span className={classes['sidebar__brand__name']}>Edward Cobbold</span>
        </Link>
        <span className={classes['sidebar__brand__job-title']}>Frontend Developer</span>

        <hr className={classes['sidebar__divider']} />

        <div className={classes['sidebar__btn-sitenav-container']}>
          <button
            type="button"
            className={classes['sidebar__btn-sitenav']}
            onClick={toggleMobileNavigation}
            aria-expanded={isMobileNavigationOpen}
          >
            <span className={classes['burger-bar']} />
            <ScreenreaderOnlyText text="Toggle menu" />
          </button>
        </div>

        <div
          className={classnames(
            classes['sidebar__sitenav-wrapper'],
            classes['sidebar__sitenav-wrapper--desktop'],
          )}
        >
          <Sitenav />
        </div>
      </div>
    </Container>

    <menu
      className={classnames(
        classes['sidebar__sitenav-wrapper'],
        classes['sidebar__sitenav-wrapper--mobile'], {
          [classes['sidebar__sitenav-wrapper--mobile--is--closed']]: !isMobileNavigationOpen,
        },
      )}
    >
      <Container>
        <Sitenav />
      </Container>
    </menu>
  </aside>
);

Sidebar.propTypes = {
  isMobileNavigationOpen: PropTypes.bool.isRequired,
  toggleMobileNavigation: PropTypes.func.isRequired,
};

export default Sidebar;
