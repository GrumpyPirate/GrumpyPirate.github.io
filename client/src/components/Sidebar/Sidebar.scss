@import 'core';

.sidebar {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  width: 100%;
  background-color: palette(theme-accent-1);
  text-transform: uppercase;
  color: palette(theme-light);
  z-index: $sidebar-zindex;

  // lg up
  @include respond-up(lg) {
    position: relative;
    height: 100vh;
    overflow-y: auto;
    touch-action: pan-y;
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    flex: 0 0 $sidebar-width-lg;
    padding: 0;
    border: 0;
    text-align: right;
  }

  // xl up
  @include respond-up(xl) {
    flex-basis: $sidebar-width;
  }

  &__content {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-top: $grid-gutter-width-base / 4;
    padding-bottom: $grid-gutter-width-base / 4;
    background-color: palette(theme-accent-1);
    height: $sidebar-height-mobile;
    z-index: $sidebar-zindex;

    // lg up
    @include respond-up(lg) {
      display: block;
      padding-top: 0;
      padding-bottom: 0;
      background: 0;
      height: auto;
    }
  }

  .sitenav__nav-item {
    // md only
    @include respond-only(md) {
      display: inline-block;

      &:not(:last-child) {
        margin-right: 1rem;
      }
    }
  }

  // Brand (Logo/Text)
  // -----------------------------------------------------------------------------------------------------------------
  &__brand {
    @include heading(5);
    margin-bottom: 0;
    letter-spacing: normal;

    // lg up
    @include respond-up(lg) {
      font-size: $font-size-h4;
    }

    // Elements
    &__image {
      display: inline-block;
      vertical-align: middle;
      width: rem(32px);
      height: rem(32px);
      margin: 0;
      background: url('./head-sprite.png') top left no-repeat transparent;
      background-size: rem(32px) rem(64px);
      image-rendering: pixelated;
      filter: drop-shadow(#{$filter-shadow-right-down-sm});
      transition: transform $anim-dur;

      // lg up
      @include respond-up(lg) {
        display: block;
        margin-left: auto;
        width: rem(48px);
        height: rem(48px);
        background-size: rem(48px) rem(96px);
      }
    }

    &__name {
      display: inline-block;
      vertical-align: middle;
      color: inherit;
      margin-left: 1rem;
    }

    &__job-title {
      display: block;
      font-size: .75em;
      color: palette(theme-light);
      opacity: .8;

      @include respond-down(md) {
        display: none;
      }
    }

    // Modifiers
    &,
    &:hover,
    &:focus {
      color: palette(theme-light);
    }

    &:hover,
    &:focus {
      .sidebar__brand__image {
        background-position: bottom left;
      }
    }

    &:active {
      .sidebar__brand__image {
        transform: scale(#{1 - (1 / 8)});
        transition-duration: 0s;
      }
    }
  }

  &__divider {
    border-style: solid none none;
    border-color: transparentize(palette(theme-light), .8);
    border-width: 1px;
    margin: 2rem 0;

    @include respond-down(md) {
      display: none;
    }
  }

  // Menu
  // -------------------------------------------------------------------------------------------------
  &__btn-sitenav-container {
    @include respond-up(md) {
      display: none;
    }
  }

  &__btn-sitenav {
    $burger-bar-width: 1.75rem;
    $burger-bar-height: 1px;
    $burger-bar-spacing: 7px;
    @include button-reset;
    @include no-outline;
    position: relative;
    display: block;
    width: $burger-bar-width;
    height: ($burger-bar-height * 3) + ($burger-bar-spacing * 2);
    transition: transform ($anim-dur / 2) ease;

    // Clicked
    &:active {
      transform: scale(#{1 - (1 / 10)});
    }

    // Menu active
    &[aria-expanded='true'] {
      .burger-bar {
        background-color: transparent;

        &::before {
          transform: rotate(-135deg);
        }

        &::after {
          transform: rotate(135deg);
        }
      }
    }

    .burger-bar {
      top: calc(50% - #{($burger-bar-height / 2)});

      &,
      &::before,
      &::after {
        display: block;
        width: $burger-bar-width;
        height: $burger-bar-height;
        position: absolute;
        left: 0;
        background-color: palette(theme-light);
        transition: transform $anim-dur ease,
                    background-color $anim-dur ease;
      }

      &::before,
      &::after {
        content: '';
      }

      &::before {
        transform: translateY(-#{$burger-bar-height + $burger-bar-spacing});
      }

      &::after {
        transform: translateY(#{$burger-bar-height + $burger-bar-spacing});
      }
    }
  }

  &__sitenav-wrapper {
    &--desktop {
      @include respond-down(sm) {
        display: none;
      }
    }

    &--mobile {
      position: fixed;
      top: $sidebar-height-mobile;
      left: 0;
      width: 100%;
      margin: 0;
      padding: .5rem 0;
      background-color: palette(theme-accent-1);
      border-top: solid $hr-border-width transparentize($sidebar-border-color, .75);
      border-bottom: solid $hr-border-width $sidebar-border-color;
      z-index: $sidebar-zindex - 1;
      transition: opacity $anim-dur ease,
                  transform $anim-dur ease,
                  visibility $anim-dur ease;

      &.is--closed {
        opacity: 0;
        visibility: hidden;
        transform: translateY(-50%);
        pointer-events: none;
      }

      @include respond-up(md) {
        display: none;
      }
    }
  }
}
