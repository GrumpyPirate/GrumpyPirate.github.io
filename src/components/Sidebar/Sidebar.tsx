import React, { FunctionComponent, useEffect, useState } from 'react';
import { Link, useLocation } from 'react-router-dom';
import styled from 'styled-components';

import ScreenreaderOnlyText from 'components/Accessibility/ScreenreaderOnlyText/ScreenreaderOnlyText';
import { Container } from 'components/Grid';
import Sitenav from 'components/Sitenav/Sitenav';
import usePrevious from 'hooks/usePrevious';
import { mediaQueries, palette, rem, sidebarWidth, sidebarWidthXl } from 'styles';
import { PropsWithClassName } from 'types/common';

import {
  Brand,
  BrandLogo,
  BrandName,
  BurgerButton,
  BurgerButtonBar,
  BurgerButtonWrapper,
  Content,
  DesktopSitenavWrapper,
  Divider,
  JobTitle,
  MobileSitenavWrapper,
} from './Sidebar.constants';

const Sidebar: FunctionComponent<PropsWithClassName> = function ({ className }) {
  const [isTouchNavigationOpen, setIsTouchNavigationOpen] = useState(false);
  const { pathname } = useLocation();
  const prevPathname = usePrevious(pathname);

  /**
   * Use a side-effect to:
   *  - Close the touch navigation, when the route changes
   */
  useEffect(() => {
    if (isTouchNavigationOpen && prevPathname !== pathname) {
      setIsTouchNavigationOpen(false);
    }
  }, [isTouchNavigationOpen, pathname, prevPathname]);

  return (
    <aside className={className}>
      <Container>
        <Content>
          <Brand>
            <Link to="/">
              <BrandLogo />
              <BrandName>Stephanie Cobbold</BrandName>
            </Link>
          </Brand>
          <JobTitle>Frontend Developer</JobTitle>

          <Divider />

          <BurgerButtonWrapper>
            <BurgerButton
              type="button"
              onClick={() => {
                setIsTouchNavigationOpen(!isTouchNavigationOpen);
              }}
              aria-expanded={isTouchNavigationOpen}
            >
              <BurgerButtonBar />
              <ScreenreaderOnlyText text="Toggle menu" />
            </BurgerButton>
          </BurgerButtonWrapper>

          <DesktopSitenavWrapper>
            <Sitenav />
          </DesktopSitenavWrapper>
        </Content>
      </Container>

      <MobileSitenavWrapper isOpen={isTouchNavigationOpen}>
        <Container>
          <Sitenav />
        </Container>
      </MobileSitenavWrapper>
    </aside>
  );
};

export default styled(Sidebar)`
  background-color: ${palette.themeMainBrand};
  color: ${palette.themeLightShades};
  left: 0;
  position: fixed;
  right: 0;
  text-transform: uppercase;
  top: 0;
  width: 100%;
  z-index: 11;

  @media ${mediaQueries.lg} {
    align-items: center;
    border: 0;
    display: flex;
    flex-wrap: wrap;
    height: 100vh;
    padding: 0;
    position: fixed;
    text-align: right;
    width: ${rem(sidebarWidth)};
  }

  @media ${mediaQueries.xl} {
    width: ${rem(sidebarWidthXl)};
  }
`;
