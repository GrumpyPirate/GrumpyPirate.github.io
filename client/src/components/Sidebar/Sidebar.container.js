import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { toggleMobileNavigation } from 'actions/ui/ui';

import Sidebar from './Sidebar.component';

const mapStateToProps = state => ({
  isMobileNavigationOpen: state.ui.isMobileNavigationOpen,
});

const mapDispatchToProps = { toggleMobileNavigation };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Sidebar));
