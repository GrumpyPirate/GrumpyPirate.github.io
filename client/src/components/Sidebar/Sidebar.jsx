import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import Sitenav from 'containers/Sitenav/Sitenav';

import './Sidebar.scss';

const Sidebar = ({ isMobileNavigationOpen, toggleMobileNavigation }) => (
  <aside className="sidebar text-lg-right">
    <div className="sidebar__content container-fluid">
      <Link to="/" className="sidebar__brand">
        <figure className="sidebar__brand__image" alt="" width="52" height="52" />
        <span className="sidebar__brand__name ml-1">Edward Cobbold</span>
      </Link>
      <span className="sidebar__brand__job-title hidden-md-down">Frontend Developer</span>

      <hr className="hidden-md-down" />

      <div className="hidden-md-up">
        <button
          type="button"
          className="sidebar__btn-sitenav"
          onClick={toggleMobileNavigation}
          aria-expanded={isMobileNavigationOpen}
        >
          <span className="burger-bar" />
          <span className="sr-only">Toggle menu</span>
        </button>
      </div>

      <div className="hidden-sm-down">
        <Sitenav />
      </div>
    </div>

    <menu
      className={classnames('sidebar__menu hidden-md-up', {
        'is--closed': !isMobileNavigationOpen,
      })}
    >
      <div className="container-fluid">
        <Sitenav />
      </div>
    </menu>
  </aside>
);

Sidebar.propTypes = {
  isMobileNavigationOpen: PropTypes.bool.isRequired,
  toggleMobileNavigation: PropTypes.func.isRequired,
};

export default Sidebar;
