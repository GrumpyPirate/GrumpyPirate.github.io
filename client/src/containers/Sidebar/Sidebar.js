import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { toggleMobileNavigation } from 'actions/ui/ui';

import Sidebar from 'components/Sidebar/Sidebar';

const mapStateToProps = state => ({
  isMobileNavigationOpen: state.ui.isMobileNavigationOpen,
});

const mapDispatchToProps = { toggleMobileNavigation };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Sidebar));
