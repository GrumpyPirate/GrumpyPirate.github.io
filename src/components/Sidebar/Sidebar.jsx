// React
import React from 'react'

// Routing
import { NavLink, Link } from 'react-router-dom'

// Components
import Sitenav from 'components/Sitenav/Sitenav.jsx'

// SCSS
import './Sidebar.scss'

// Images
import Logo from 'images/avatar-ed--head-only.png'

class Sidebar extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      navMenuOpen: false
    }

    // Lexical this binding
    this.toggleMenu = this.toggleMenu.bind(this)
  } // /constructor(props)

  toggleMenu() {
    this.setState({
      navMenuOpen: !this.state.navMenuOpen
    })
  } // /openMenu()

  render() {
    const CREATION_YEAR = 2017
    const CURR_YEAR     = new Date().getFullYear()
    const navMenuOpen   = this.state.navMenuOpen

    return (
      <aside className="sidebar text-lg-right">
        <div className="sidebar__content container-fluid">
          <Link to="/" className="sidebar__brand">
            <img className="sidebar__brand__image" src={Logo} alt="" width="52" height="52"/>
            {/*<span className="sidebar__brand__name ml-1">Edward Cobbold</span>*/}
            <span className="sidebar__brand__name ml-1">Lorem Ipsum</span>
          </Link>
          {/*<span className="sidebar__brand__job-title hidden-md-down">Frontend Developer</span>*/}
          <span className="sidebar__brand__job-title hidden-md-down">Culpa similique</span>

          <hr className="hidden-md-down"/>

          {/* Mobile nav */}
          <div className="hidden-md-up">
            <button type="button" className="sidebar__btn-sitenav" onClick={this.toggleMenu} aria-expanded={navMenuOpen}>
              <span className="burger-bar"></span>
              <span className="sr-only">Toggle menu</span>
            </button>

            <menu className="sidebar__menu" hidden={!navMenuOpen}>
              <div className="container-fluid">
                <Sitenav/>
              </div>
            </menu>
          </div>

          {/* Tablet+ nav */}
          <div className="hidden-sm-down">
            <Sitenav/>
          </div>
        </div>

        <div className="sidebar__credits hidden-md-down">
          <div className="container-fluid">
            <p>
              <small>&copy; {CURR_YEAR > CREATION_YEAR ?
                <span className="sidebar__credits__date">{CREATION_YEAR}&mdash;{CURR_YEAR}</span> :
                <span className="sidebar__credits__date">{CURR_YEAR}</span>
              }</small>
            </p>
          </div>
        </div>
      </aside>
    )
  } // /render()
} // /export default class Sidebar extends React.Component

export default Sidebar
