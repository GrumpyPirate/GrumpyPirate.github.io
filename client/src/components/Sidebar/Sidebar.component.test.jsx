import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from './Sidebar.component';

import fixture from './Sidebar.component.fixture';

describe('Components', () => {
  describe('Sidebar', () => {
    const props = {
      ...fixture,
      toggleMobileNavigation: jest.fn(),
    };

    it('should be defined', () => {
      expect(Sidebar).toBeDefined();
    });

    it('should be a function', () => {
      expect(typeof Sidebar).toBe('function');
    });

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<Sidebar {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
