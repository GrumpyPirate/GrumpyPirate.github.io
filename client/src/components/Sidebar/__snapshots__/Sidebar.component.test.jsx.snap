// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Sidebar should render as expected, without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    isMobileNavigationOpen={false}
    toggleMobileNavigation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Container>
          <div
            className="sidebar__content"
          >
            <Link
              className="sidebar__brand"
              replace={false}
              to="/"
            >
              <figure
                alt=""
                className="sidebar__brand__image"
                height="52"
                width="52"
              />
              <span
                className="sidebar__brand__name"
              >
                Edward Cobbold
              </span>
            </Link>
            <span
              className="sidebar__brand__job-title"
            >
              Frontend Developer
            </span>
            <hr
              className="sidebar__divider"
            />
            <div
              className="sidebar__btn-sitenav-container"
            >
              <button
                aria-expanded={false}
                className="sidebar__btn-sitenav"
                onClick={[MockFunction]}
                type="button"
              >
                <span
                  className="burger-bar"
                />
                <ScreenreaderOnlyText
                  text="Toggle menu"
                />
              </button>
            </div>
            <div
              className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop"
            >
              <withRouter(Connect(Sitenav)) />
            </div>
          </div>
        </Container>,
        <menu
          className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--mobile sidebar__sitenav-wrapper--mobile--is--closed"
        >
          <Container>
            <withRouter(Connect(Sitenav)) />
          </Container>
        </menu>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="sidebar__content"
          >
            <Link
              className="sidebar__brand"
              replace={false}
              to="/"
            >
              <figure
                alt=""
                className="sidebar__brand__image"
                height="52"
                width="52"
              />
              <span
                className="sidebar__brand__name"
              >
                Edward Cobbold
              </span>
            </Link>
            <span
              className="sidebar__brand__job-title"
            >
              Frontend Developer
            </span>
            <hr
              className="sidebar__divider"
            />
            <div
              className="sidebar__btn-sitenav-container"
            >
              <button
                aria-expanded={false}
                className="sidebar__btn-sitenav"
                onClick={[MockFunction]}
                type="button"
              >
                <span
                  className="burger-bar"
                />
                <ScreenreaderOnlyText
                  text="Toggle menu"
                />
              </button>
            </div>
            <div
              className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop"
            >
              <withRouter(Connect(Sitenav)) />
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="sidebar__brand"
                replace={false}
                to="/"
              >
                <figure
                  alt=""
                  className="sidebar__brand__image"
                  height="52"
                  width="52"
                />
                <span
                  className="sidebar__brand__name"
                >
                  Edward Cobbold
                </span>
              </Link>,
              <span
                className="sidebar__brand__job-title"
              >
                Frontend Developer
              </span>,
              <hr
                className="sidebar__divider"
              />,
              <div
                className="sidebar__btn-sitenav-container"
              >
                <button
                  aria-expanded={false}
                  className="sidebar__btn-sitenav"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <span
                    className="burger-bar"
                  />
                  <ScreenreaderOnlyText
                    text="Toggle menu"
                  />
                </button>
              </div>,
              <div
                className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop"
              >
                <withRouter(Connect(Sitenav)) />
              </div>,
            ],
            "className": "sidebar__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <figure
                    alt=""
                    className="sidebar__brand__image"
                    height="52"
                    width="52"
                  />,
                  <span
                    className="sidebar__brand__name"
                  >
                    Edward Cobbold
                  </span>,
                ],
                "className": "sidebar__brand",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "sidebar__brand__image",
                    "height": "52",
                    "width": "52",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "figure",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edward Cobbold",
                    "className": "sidebar__brand__name",
                  },
                  "ref": null,
                  "rendered": "Edward Cobbold",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Frontend Developer",
                "className": "sidebar__brand__job-title",
              },
              "ref": null,
              "rendered": "Frontend Developer",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sidebar__divider",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  aria-expanded={false}
                  className="sidebar__btn-sitenav"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <span
                    className="burger-bar"
                  />
                  <ScreenreaderOnlyText
                    text="Toggle menu"
                  />
                </button>,
                "className": "sidebar__btn-sitenav-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-expanded": false,
                  "children": Array [
                    <span
                      className="burger-bar"
                    />,
                    <ScreenreaderOnlyText
                      text="Toggle menu"
                    />,
                  ],
                  "className": "sidebar__btn-sitenav",
                  "onClick": [MockFunction],
                  "type": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "burger-bar",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "text": "Toggle menu",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <withRouter(Connect(Sitenav)) />,
                "className": "sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Container>
            <withRouter(Connect(Sitenav)) />
          </Container>,
          "className": "sidebar__sitenav-wrapper sidebar__sitenav-wrapper--mobile sidebar__sitenav-wrapper--mobile--is--closed",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withRouter(Connect(Sitenav)) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "menu",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Container>
            <div
              className="sidebar__content"
            >
              <Link
                className="sidebar__brand"
                replace={false}
                to="/"
              >
                <figure
                  alt=""
                  className="sidebar__brand__image"
                  height="52"
                  width="52"
                />
                <span
                  className="sidebar__brand__name"
                >
                  Edward Cobbold
                </span>
              </Link>
              <span
                className="sidebar__brand__job-title"
              >
                Frontend Developer
              </span>
              <hr
                className="sidebar__divider"
              />
              <div
                className="sidebar__btn-sitenav-container"
              >
                <button
                  aria-expanded={false}
                  className="sidebar__btn-sitenav"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <span
                    className="burger-bar"
                  />
                  <ScreenreaderOnlyText
                    text="Toggle menu"
                  />
                </button>
              </div>
              <div
                className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop"
              >
                <withRouter(Connect(Sitenav)) />
              </div>
            </div>
          </Container>,
          <menu
            className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--mobile sidebar__sitenav-wrapper--mobile--is--closed"
          >
            <Container>
              <withRouter(Connect(Sitenav)) />
            </Container>
          </menu>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="sidebar__content"
            >
              <Link
                className="sidebar__brand"
                replace={false}
                to="/"
              >
                <figure
                  alt=""
                  className="sidebar__brand__image"
                  height="52"
                  width="52"
                />
                <span
                  className="sidebar__brand__name"
                >
                  Edward Cobbold
                </span>
              </Link>
              <span
                className="sidebar__brand__job-title"
              >
                Frontend Developer
              </span>
              <hr
                className="sidebar__divider"
              />
              <div
                className="sidebar__btn-sitenav-container"
              >
                <button
                  aria-expanded={false}
                  className="sidebar__btn-sitenav"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <span
                    className="burger-bar"
                  />
                  <ScreenreaderOnlyText
                    text="Toggle menu"
                  />
                </button>
              </div>
              <div
                className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop"
              >
                <withRouter(Connect(Sitenav)) />
              </div>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="sidebar__brand"
                  replace={false}
                  to="/"
                >
                  <figure
                    alt=""
                    className="sidebar__brand__image"
                    height="52"
                    width="52"
                  />
                  <span
                    className="sidebar__brand__name"
                  >
                    Edward Cobbold
                  </span>
                </Link>,
                <span
                  className="sidebar__brand__job-title"
                >
                  Frontend Developer
                </span>,
                <hr
                  className="sidebar__divider"
                />,
                <div
                  className="sidebar__btn-sitenav-container"
                >
                  <button
                    aria-expanded={false}
                    className="sidebar__btn-sitenav"
                    onClick={[MockFunction]}
                    type="button"
                  >
                    <span
                      className="burger-bar"
                    />
                    <ScreenreaderOnlyText
                      text="Toggle menu"
                    />
                  </button>
                </div>,
                <div
                  className="sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop"
                >
                  <withRouter(Connect(Sitenav)) />
                </div>,
              ],
              "className": "sidebar__content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <figure
                      alt=""
                      className="sidebar__brand__image"
                      height="52"
                      width="52"
                    />,
                    <span
                      className="sidebar__brand__name"
                    >
                      Edward Cobbold
                    </span>,
                  ],
                  "className": "sidebar__brand",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "sidebar__brand__image",
                      "height": "52",
                      "width": "52",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "figure",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Edward Cobbold",
                      "className": "sidebar__brand__name",
                    },
                    "ref": null,
                    "rendered": "Edward Cobbold",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Frontend Developer",
                  "className": "sidebar__brand__job-title",
                },
                "ref": null,
                "rendered": "Frontend Developer",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "sidebar__divider",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    aria-expanded={false}
                    className="sidebar__btn-sitenav"
                    onClick={[MockFunction]}
                    type="button"
                  >
                    <span
                      className="burger-bar"
                    />
                    <ScreenreaderOnlyText
                      text="Toggle menu"
                    />
                  </button>,
                  "className": "sidebar__btn-sitenav-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-expanded": false,
                    "children": Array [
                      <span
                        className="burger-bar"
                      />,
                      <ScreenreaderOnlyText
                        text="Toggle menu"
                      />,
                    ],
                    "className": "sidebar__btn-sitenav",
                    "onClick": [MockFunction],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "burger-bar",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "text": "Toggle menu",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <withRouter(Connect(Sitenav)) />,
                  "className": "sidebar__sitenav-wrapper sidebar__sitenav-wrapper--desktop",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Container>
              <withRouter(Connect(Sitenav)) />
            </Container>,
            "className": "sidebar__sitenav-wrapper sidebar__sitenav-wrapper--mobile sidebar__sitenav-wrapper--mobile--is--closed",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <withRouter(Connect(Sitenav)) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "menu",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
